### PR TITLE
Fix port evaluation for machine exec

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Fix port evaluation for machine exec.

### Screenshot/screencast of this PR
**Before**:
Screenshot_20210514_143215

**After**:
Screenshot_20210514_160551

### What issues does this PR fix or reference?
it's for https://github.com/eclipse/che/issues/19810

### How to test this PR?
1. Create devworkspace like https://github.com/sleshchenko/NodeJS-Sample-App/blob/main/devfile.yaml
2. Try to execute the command

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

**Comment by azatsarynnyy** [comment]
@sleshchenko you need to rebase your branch onto the origin's master as it contains the commit that fixes the build

File: plugins/task-plugin/src/che-workspace-client.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export class CheWorkspaceClient {
 
   async getMachineExecServerURL(): Promise<string> {
     const machineExecEndpoint = await this.getMachineExecEndpoint();
-    const port = machineExecEndpoint.attributes?.port || machineExecEndpoint.attributes?.targetPort;
+    const port = machineExecEndpoint.attributes?.port || machineExecEndpoint.targetPort;
```
Comment (benoitf) [inline plugins/task-plugin/src/che-workspace-client.ts:48]: from what I understand, if you map in `CheServerEndpointServiceImpl` the property targetPort
then here we could just use `machineExecEndpoint.targetPort` all the time no ?

Comment (sleshchenko) [reply plugins/task-plugin/src/che-workspace-client.ts:48]: Yeap, it's here from the time I went another way.
I'll try to remove it

**Comment by sleshchenko** [comment]
[crw-ci-test]

**Comment by sleshchenko** [comment]
I checked manually that terminal and tasks works on quay.io/crw_pr/che-theia:1102 built for happy path test
Che Workspace
Screenshot_20210521_170821
DevWorkspace
Screenshot_20210521_171210

**Review by benoitf — APPROVED** [review]

**Pull request merged**